src/ion/requirement.rs

```diff
@@ -31,39 +32,53 @@ impl Requirement {
             (Requirement::Unknown, other) | (other, Requirement::Unknown) => other,
             (Requirement::Conflict, _) | (_, Requirement::Conflict) => Requirement::Conflict,
             (other, Requirement::Any) | (Requirement::Any, other) => other,
-            (Requirement::Stack, Requirement::Stack) => self,
-            (Requirement::Register, Requirement::Fixed(preg))
-            | (Requirement::Fixed(preg), Requirement::Register) => Requirement::Fixed(preg),
             (Requirement::Register, Requirement::Register) => self,
-            (Requirement::Fixed(a), Requirement::Fixed(b)) if a == b => self,
+            (Requirement::Stack, Requirement::Stack) => self,
+            (Requirement::Register, Requirement::FixedReg(preg))
+            | (Requirement::FixedReg(preg), Requirement::Register) => Requirement::FixedReg(preg),
+            (Requirement::Stack, Requirement::FixedStack(preg))
+            | (Requirement::FixedStack(preg), Requirement::Stack) => Requirement::FixedStack(preg),
+            (Requirement::FixedReg(a), Requirement::FixedReg(b)) if a == b => self,
+            (Requirement::FixedStack(a), Requirement::FixedStack(b)) if a == b => self,
             _ => Requirement::Conflict,
         }
     }
+}
+
+impl<'a, F: Function> Env<'a, F> {
     #[inline(always)]
-    pub fn from_operand(op: Operand) -> Requirement {
+    pub fn requirement_from_operand(&self, op: Operand) -> Requirement {
         match op.constraint() {
-            OperandConstraint::FixedReg(preg) => Requirement::Fixed(preg),
+            OperandConstraint::FixedReg(preg) => {
+                if self.pregs[preg.index()].is_stack {
+                    Requirement::FixedStack(preg)
+                } else {
+                    Requirement::FixedReg(preg)
+                }
+            }
             OperandConstraint::Reg | OperandConstraint::Reuse(_) => Requirement::Register,
             OperandConstraint::Stack => Requirement::Stack,
             _ => Requirement::Any,
         }
     }
-}
 
-impl<'a, F: Function> Env<'a, F> {
-    pub fn compute_requirement(&self, bundle: LiveBundleIndex) -> Requirement {
+    pub fn compute_requirement(&self, bundle: LiveBundleIndex) -> (Requirement, ProgPoint) {
```

Comment:
A doc-comment describing the return value here would be useful. Actually, a struct might be nice: something like

```
struct RequirementResult {
    pub req: Requirement,
    pub needs_split: Option<ProgPoint>,
}
```

for readability maybe?

Reply:
(Good idea to return the splitpoint eagerly rather than re-scanning to find it later, by the way -- every little efficiency improvement counts!)